aws-api/src/main/java/com/amplifyframework/api/aws/operation/AWSRestOperation.java

```diff
@@ -113,13 +116,20 @@ public void onResponse(@NonNull Call call,
                                @NonNull Response response) throws IOException {
             final ResponseBody responseBody = response.body();
             final int statusCode = response.code();
-
+            final Map<String, String> headersMap = new HashMap<>();
+            final Map<String, List<String>> headersMultiMap = response.headers().toMultimap();
+            for (String key : headersMultiMap.keySet()) {
+                List<String> value = headersMultiMap.get(key);
+                if (value.size() > 0) {
+                    headersMap.put(key, value.get(0));
```

Comment:
Well, I'll be derned. RFC2616 (for HTTP/1.1) [allows multiple values per key](https://stackoverflow.com/a/24502264/695787).

It might be easy enough to transform this to something compliant with the RFC. I probably _wouldn't_ worry about this, if it weren't just a quick update to your existing code:

```suggestion
                    headersMap.put(key, TextUtils.join(",", value));
```

Reply:
Yep, I considered returning a `Map<String, List<String>>`.   from `RestResponse.getHeaders()`.  But then I thought, Amplify is _opinionated_, and I believe our opinion should be that you **shouldn't** use multiple headers with the same key.

That said, I actually this suggestion!  It allows us to return a simple `Map<String, String>` for almost all use cases, but doesn't block customers from accessing all the headers if they happen to have multiple.

Reply:
I've made this change, and also updated the unit test to verify it.